### ISSUE TITLE
Wrong matchesPosition 'start' for Cyrillic text.

### ISSUE THREAD

**Issue by shanginn** [meilisearch/meilisearch#3938]
First of all thank you for this wonderful piece of tech! Such a delight to work with.
And second, I'm only begin to work with it, so it might be incorrect configuration from my part, please point me in right direction if it is so.

The problem example:
title == `Разработка заданий на развитие функциональной грамотности по теме "Африка"`
query == `Африка`
matchesPosition.title.0 == `{start: 125, length: 6}`

So the length is correct, but the start is not. Whole title is 74 characters long. If the matching query at the beginning of the string - all good, start is 0.

And other thing: `_formatted` response is highlighting correct part of the string:
`Разработка заданий на развитие функциональной грамотности по теме "<ais-highlight-0000000000>Африка</ais-highlight-0000000000>"`

I'm using getmeili/meilisearch:latest inside *docker compose* with [symfony bundle](https://github.com/meilisearch/meilisearch-symfony).

And I only assume that the problem with Cyrillic texts, because it is the only kind of texts I have.

Thank you and have a wonderful day

**Comment by Kerollmops** [comment]
Hey @shanginn 👋

Thank you for your investigation. Meilisearch seems to be returning a byte-scaled `matchesPosition::start` field but a char-scaled `matchesPosition::length` field. According to the documentation, the `length` must be expressed in bytes and no chars.

For now, you can use the `length` like a char-scaled field instead of bytes like the `start`.

Have a good day 🥥

**Comment by shanginn** [comment]
hello good sir. thank you for the response.
so it is a bug, but not in the `start`, but in the `length`? I thought `start` is suppose to be char-scaled too.

Ok, for now I'm using `_formatted` version, but will keep in mind for the future that both `start` and `length` are bytes values.

**Comment by ManyTheFish** [comment]
Yes, the match positions are in bytes, below is the link to the documentation:
https://www.meilisearch.com/docs/reference/api/search#show-matches-position

**Comment by ManyTheFish** [comment]
Hello @jk-recommendy,

I updated the PR as you asked, I hope it is clearer :D

Have a good evening!

**Comment by MattExact** [comment]
I'm encountering errors with highlighting strings containing characters such as  `"é"` / `b'\xc3\xa9'` as it's being returned as length 1, I think it's due to this issue.

Is there any update on this? @Kerollmops / @ManyTheFish 
The PR seems to have been shelved and been forgotten about.

**Comment by ManyTheFish** [comment]
Hello @MattExact,
Yes, you're right; it has been shelved. I honestly don't have that much time to reproduce and investigate your case,

Could you please provide an exact reproduction of the bug with curl requests?

This will save me a lot of time to investigate the bug and propose a fix,

I'm sorry for the delay in fixing the bug... 😞

**Comment by MattExact** [comment]
> Could you please provide an exact reproduction of the bug with curl requests?

@ManyTheFish It was easier to write in Python, hope that's not a problem
 
```python
from meilisearch import Client

client = Client("http://localhost:7700")
index = "showMatchesPosition_test"

create_index = client.create_index(index, {"primaryKey":"id"})
client.wait_for_task(create_index.task_uid)
add_text = client.index(index).add_documents(
    [
        {"id": 1, "text": "£100 per month"},
        {"id": 2, "text": "Beyoncé Giselle Knowles-Carter reviewed Théo's résumé."},
    ]
)
client.wait_for_task(add_text.task_uid)

results_1 = client.index(index).search("£100", {"showMatchesPosition":True})
print(results_1["hits"][0]["_matchesPosition"]) # -> {'text': [{'start': 0, 'length': 4}]}
print(len("£100".encode())) # <- length 5 - _matchesPosition gives length 4


results_2 = client.index(index).search("beyonce theo", {"showMatchesPosition": True})
print(results_2["hits"][0]["_matchesPosition"]) # -> {'text': [{'start': 0, 'length': 7}, {'start': 41, 'length': 4}]}
print(len("Beyoncé".encode())) # <- length 8 - _matchesPosition gives length 7
print(len("Beyoncé Giselle Knowles-Carter reviewed ".encode())) # start is correctly given as 41
print(len("Théo".encode())) # <- length 5 - _matchesPosition gives length 4
```

**Comment by MattExact** [comment]
I think this may now be fixed by https://github.com/meilisearch/meilisearch/pull/5446
See also: https://github.com/meilisearch/meilisearch/issues/5429
